### PR TITLE
Add notes about Adyen App refunds logic

### PR DESCRIPTION
**Pull request by witoszekdev**
Added notes how refund `lineItems` are handled depending on used refund type.

### CODE REVIEW

**Review by krzysztofwolski — APPROVED** [review]
(Non blocking) Is there any reason we should recommend one way over another? If so, would appreciate adding it to the docs.

File: docs/developer/app-store/apps/adyen/architecture.mdx [hunk 1]
```diff
@@ -32,3 +38,26 @@ If Adyen surpasses this limit, the App will return a FAILURE status with an appr
...
 
 If such timeout happens, the created [`TransactionItem`](api-reference/payments/objects/transaction-item.mdx) will not have the metadata from Adyen on `additionalDetails` object, which includes payment method type, credit card brand, etc.
+
+### Granted refunds vs manual refunds
+
+There are differences in how refunds are handled depending on wether an [`OrderGrantedRefund`](/api-reference/orders/objects/order-granted-refund) was created in Saleor and a refund via [`transactionRequestRefundForGrantedRefund`](api-reference/payments/mutations/transaction-request-refund-for-granted-refund) mutation was requested, or a "manual refund" sent via [`transactionRequestAction`](/api-reference/payments/mutations/transaction-request-action) mutation was requested. The differences is in how [`lineItems`](https://docs.adyen.com/api-explorer/Checkout/71/post/payments/_paymentPspReference_/refunds#request-lineItems) are reported to Adyen
```
Comment (aniav) [inline docs/developer/app-store/apps/adyen/architecture.mdx:44]: The links to our docs should always have `.mdx` at the end. Other formats can cause Docusaurus to break when moving files around.

File: docs/developer/app-store/apps/adyen/architecture.mdx [hunk 2]
```diff
@@ -32,3 +38,26 @@ If Adyen surpasses this limit, the App will return a FAILURE status with an appr
...
+
+
+ <Badge
+   text="Added in Saleor 3.15"
+   class="badge badge--secondary margin-bottom--sm"
+ />
+
+A granted refund is initiated using the [`transactionRequestRefundForGrantedRefund`](/api-reference/payments/mutations/transaction-request-refund-for-granted-refund) mutation.
+
+When granted refund is requested, app will map `grantedRefund.lines.orderLines` to `lineItems` reported to Adyen. Additionally, if `grantedRefund.shippingCostIncluded`` is set to true, the app will include a shipping line in the lineItems sent to Adyen.
```
Comment (aniav) [inline docs/developer/app-store/apps/adyen/architecture.mdx:63]:
```suggestion
When granted refund is requested, app will map `grantedRefund.lines.orderLines` to `lineItems` reported to Adyen. Additionally, if `grantedRefund.shippingCostIncluded` is set to true, the app will include a shipping line in the lineItems sent to Adyen.
```

File: docs/developer/app-store/apps/adyen/architecture.mdx [hunk 3]
```diff
@@ -32,3 +38,26 @@ If Adyen surpasses this limit, the App will return a FAILURE status with an appr
...
+### Granted refunds vs manual refunds
+
+There are differences in how refunds are handled depending on wether an [`OrderGrantedRefund`](/api-reference/orders/objects/order-granted-refund) was created in Saleor and a refund via [`transactionRequestRefundForGrantedRefund`](api-reference/payments/mutations/transaction-request-refund-for-granted-refund) mutation was requested, or a "manual refund" sent via [`transactionRequestAction`](/api-reference/payments/mutations/transaction-request-action) mutation was requested. The differences is in how [`lineItems`](https://docs.adyen.com/api-explorer/Checkout/71/post/payments/_paymentPspReference_/refunds#request-lineItems) are reported to Adyen
+
+#### Manual refund
+
+A manual refund is initiated using the [`transactionRequestAction`](/api-reference/payments/mutations/transaction-request-action) mutation.
+
+When manual refund is requested, the app will send `lineItems` to Adyen when the `amount` requested for refund is equal to or greater than Order's total gross amount.
+In order cases app will not send `lineItems` to Adyen.
```
Comment (aniav) [inline docs/developer/app-store/apps/adyen/architecture.mdx:51]:
```suggestion
In oder cases app will not send `lineItems` to Adyen.
```

File: docs/developer/app-store/apps/adyen/architecture.mdx [hunk 4]
```diff
@@ -32,3 +38,26 @@ If Adyen surpasses this limit, the App will return a FAILURE status with an appr
...
+
+### Granted refunds vs manual refunds
+
+There are differences in how refunds are handled depending on wether an [`OrderGrantedRefund`](/api-reference/orders/objects/order-granted-refund) was created in Saleor and a refund via [`transactionRequestRefundForGrantedRefund`](api-reference/payments/mutations/transaction-request-refund-for-granted-refund) mutation was requested, or a "manual refund" sent via [`transactionRequestAction`](/api-reference/payments/mutations/transaction-request-action) mutation was requested. The differences is in how [`lineItems`](https://docs.adyen.com/api-explorer/Checkout/71/post/payments/_paymentPspReference_/refunds#request-lineItems) are reported to Adyen
+
+#### Manual refund
+
+A manual refund is initiated using the [`transactionRequestAction`](/api-reference/payments/mutations/transaction-request-action) mutation.
+
+When manual refund is requested, the app will send `lineItems` to Adyen when the `amount` requested for refund is equal to or greater than Order's total gross amount.
```
Comment (aniav) [inline docs/developer/app-store/apps/adyen/architecture.mdx:50]:
```suggestion
When manual refund is requested, the app will send `lineItems` to Adyen for the `amount` equal to or greater than Order's total gross amount.
```

I am not sure if I understand this sentence, but I tried fixing it the way I understand it 😅

**Review by aniav — COMMENTED** [review]
A few things to fix

File: docs/developer/app-store/apps/adyen/architecture.mdx (hunk 3 above)
Comment (witoszekdev) [reply docs/developer/app-store/apps/adyen/architecture.mdx]: oder, aka [Odra](https://maps.app.goo.gl/2VGXSDbgvphAi7Gy8)? :trollface: 

I've fixed it to "other"

File: docs/developer/app-store/apps/adyen/architecture.mdx (hunk 4 above)
Comment (witoszekdev) [reply docs/developer/app-store/apps/adyen/architecture.mdx]: `amount` is input in the `transactionRequestAction` mutation, so **if** amount == `order.totalGrossAmount` then we send `lineItems`

**Review by aniav — APPROVED** [review]

**Comment by witoszekdev** [comment]
@krzysztofwolski Staff should as a rule of thumb use granted refunds, since certain payment methods like AfterPay, Klarna, etc. (described in the linked Adyen docs) do not support refunds `lineItems` is not included in Adyen API request.

**Pull request merged**